### PR TITLE
feat: Add the RemoveMicrosoftApplicationInsightsLoggerProvider extension to all Serilog infrastructure code

### LINKED ISSUE #604: Use Observability's `RemoveMicrosoftApplicationInsightsLoggerProvider` extension in Azure Functions project templates
**Is your feature request related to a problem? Please describe.**
When looking at improving the Application Insights telemetry in Azure Functions, we saw that we should remove the default Microsoft's telemetry as it conflicts with Arcus' telemetry. Therefore, the Observability library has created an `.RemoveMicrosoftApplicationInsightsLoggerProvider` extension that we can use when setting up our Serilog infrastructure.

**Describe the solution you'd like**
Add the `RemoveMicrosoftApplicationInsightsLoggerProvider` extension to all Serilog infrastructure code in all Azure Functions project templates.

**Additional context**
Relates to https://github.com/arcus-azure/arcus.observability/issues/364

### PR DESCRIPTION
**Pull request by vtermaat**
Add the RemoveMicrosoftApplicationInsightsLoggerProvider extension to all Serilog infrastructure code.
Closes #604 

Replaces: #607

### CODE REVIEW

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Arcus.Templates.AzureFunctions.Databricks.JobMetrics.csproj [hunk 1]
```diff
@@ -25,7 +25,6 @@
     <ContentTargetFolders>content</ContentTargetFolders>
     <EnableDefaultContentItems>false</EnableDefaultContentItems>
     <DockerComposeProjectPath>..\Arcus.Templates.Orchestrator.dcproj</DockerComposeProjectPath>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Arcus.Templates.AzureFunctions.Databricks.JobMetrics.csproj:28]: No need anymore to remove this, as this was only needed when we used the preview version.

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 2]
```diff
@@ -41,8 +42,10 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                    .AddSerilog(logConfig.CreateLogger(), dispose: true);
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:46]: Hihi, same as before, aligning on the `.` would be better.

Comment (vtermaat) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:46]: oh that's what you were aiming for. got it!

Comment (vtermaat) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs]: https://github.com/arcus-azure/arcus.templates/pull/610/commits/d6f805e01da7787cf6826579a07d63af4d60c3ee

**Review by stijnmoreels — APPROVED** [review]
That's it! 🥳

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 3]
```diff
@@ -41,8 +42,10 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
             });
+
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:48]:
```suggestion
```

**Comment by fgheysels** [comment]
Looks like we have a failing test.  Can you have a look at it @vtermaat ?

**Comment by stijnmoreels** [comment]
Ran them again, as the build is retained.

**Pull request merged**